README.md

```diff
@@ -217,6 +217,9 @@ The default values are shown after each option key.
 	compress: true,     // support gzip/deflate content encoding. false to disable
 	size: 0,            // maximum response body size in bytes. 0 to disable
 	agent: null         // http(s).Agent instance, allows custom proxy, certificate etc.
+	lookup: dns.lookup  // custom dns lookup function for socket
+	family: null        // numbers 4 or 6 to limit dns resolution to ip v4 or v6
+	localAddress: null  // local interface to bind to for network connections
```

Comment:
I agree on adding the lookup option. 

I think the `family` option can be passed using an http agent though so we wouldn’t need to add another option for that? 

And I’m not sure what `localAddress` is needed for? Is it like bindAddress for servers? Why would a request library need it? The PR at https://github.com/bitinn/node-fetch/pull/256 does not have much info on it either. Maybe you could clarify it in the comment or add an explanation to the new node.js-only properties in the README for others who are puzzled by it?

Reply:
While trying to find anything on how to specify the address family through an agent, I found an (admittedly old) [comment about the custom lookup only being used when not using an agent](https://github.com/nodejs/node/issues/4018#issuecomment-159600574)
Maybe we should investigate the interactions of these options a bit more. I'll set up some test cases tomorrow.

I can only guess that the use case for `localAddress` might be to influence routing by selecting the outgoing interface.

Reply:
To clarify: The itch I'm scratching with this MR is really only the `lookup` option.
While implementing it I looked at other PRs and found that they touched the same code paths and would probably conflict with mine (and also not quite work as intended). So I folded them into this MR. In retrospect this probably was not as good an idea as I thought it was. I'll gladly reduce this back to just adding the lookup option and have it merged, instead of testing and reasoning about code by other people, which may not even be interested any longer.

Reply:
> While trying to find anything on how to specify the address family through an agent, I found an (admittedly old) comment about the custom lookup only being used when not using an agent.

I had a similar experience when I researched the same thing for [`@octokit/rest`](https://github.com/octokit/rest.js). It used to have the request options `ca`, `proxy`, `family` and `rejectUnauthorized`. I think I wasn’t able to make the `family` option work myself with http agent, I only was able to create tests for the other three: https://github.com/octokit/rest.js/pull/783/commits/bfec75abfeb0154fa10be8b450fe072734c50bab. 

I still use an agent for the family option but I don’t actually know if it works:
https://github.com/octokit/rest.js/blob/7e5c8a501f25df1758e4bc2c162f6e307436062f/lib/get-request-agent.js#L20 ¯\\\_(ツ)\_/¯

So yeah, I don’t know. Nobody complained about it so far. Maybe in order to move things forward just implement the lookup option and leave out the other two? I’d not add them without having a clear use case that we agree on is within the scope of `node-fetch`